### PR TITLE
Many UI tests and refactoring of activities.

### PR DESCRIPTION
**Pull request by Ishadijcks**
image

### CODE REVIEW

**Comment by Ishadijcks** [comment]
Missed some PMD errors.

**Comment by joskuijpers** [comment]
The codecov gives crosses because of a very tiny change in coverage. This is because of needed changes to make the code testable. However, the tests for GUI can't be run on travis :(
That means that instead of a 20% increase in coverage, that this PR actually does, it decreases coverage by 0.4%.

We are setting codecov as non required because of the travis issues, and merging this PR.

**Comment by joskuijpers** [comment]
We are using `//TRAVIS @Ignore` to magically ignore GUI tests on Travis so that builds succeed.
A travis build replaces this with @Ignore. When running locally, the tests will be run.

File: BankChain/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -83,9 +83,10 @@
...
     <string name="pref_private_key">Private key</string>
     <string name="menu_refresh">Refresh</string>
+    <string name="cancel">Cancel</string>
+    <string name="new_verification">New verification</string>
     <string name="public_key">Public key</string>
     <string name="iban">IBAN</string>
     <string name="verify">Verify</string>
-    <string name="cancel">Cancel</string>
-    <string name="new_verification">New verification</string>
+>>>>>>> Stashed changes
```
Comment (joskuijpers) [inline BankChain/app/src/main/res/values/strings.xml]: What is this?

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/NewVerificationActivity.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            String publicKey = publicKeyText.getText().toString();
+            String iban = ibanText.getText().toString();
+            Log.d("GUI", "onClick: Gonna verify, but not really...");
+            Log.d("GUI", "Public key: " + publicKey);
+            Log.d("GUI", "IBAN: " + iban);
+            this.publicKey = publicKey;
+            this.iban = iban;
+        });
+    }
+
```
Comment (joskuijpers) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/NewVerificationActivity.java:44]: Newline

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/NewVerificationActivity.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_new_verification);
+        initializeListeners();
+
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        if (getSupportActionBar() != null) {
+            Log.d("GUI", "onCreate: Actionbar found");
```
Comment (joskuijpers) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/NewVerificationActivity.java]: Debug? Is this necessary?

File: BankChain/app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -5,6 +5,8 @@
     <!-- Permission to talk to banks -->
     <uses-permission android:name="android.permission.INTERNET" />
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
+    <uses-permission android:name="android.permission.DISABLE_KEYGUARD"/>
```
Comment (joskuijpers) [inline BankChain/app/src/main/AndroidManifest.xml]: Are these permissions needed? We don't run on Travis.
Might be useful to add an XML comment about these two, for future reference.

Comment (Ishadijcks) [reply BankChain/app/src/main/AndroidManifest.xml]: Whew good catch,
A remnant from trying to make Travis work.

**Review by joskuijpers — APPROVED** [review]

**Pull request merged**